### DISCUSSION TITLE
color scheme per domain

### DISCUSSION THREAD

**Discussion by javaonekenobi** [wezterm/wezterm#4260] [Q&A]
Hi,
is there a way to set the color scheme depending on the domain I'm connecting to ?

thanks

**Comment by wez** [comment]
Config is per-window, but you can override it based on the active pane in the active tab:

* https://wezfurlong.org/wezterm/config/lua/window-events/update-status.html - to know when the active pane is changed in some way
* https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html - to override the config

Something like:

```lua
local wezterm = require 'wezterm'
local config = wezterm.config_builder()

-- map from the domain name to the color scheme
local DOMAIN_TO_SCHEME = {
   -- the built-in local domain is called `local`
   ["local"] = "Batman",
   -- add other entries here to match the names of your various domains
  ["my.domain"] = "Grass"
}

wezterm.on('update-status', function(window, pane)
   local overrides = window:get_config_overrides() or {}
   overrides.color_scheme = DOMAIN_TO_SCHEME[pane:get_domain_name()]
   window:set_config_overrides(overrides)  
end)

return config
```

It is not currently possible to set the config on a per-pane basis, so if you use splits in the same tab, you won't be able to see the different schemes side by side in that tab.

**Comment by javaonekenobi** [comment]
Oh, thanks a lot, I was a bit lost there, per-pane will be more than good,
I normally use splits only within the same context.

I've been looking for a good terminal for ages (was stuck to aterm until
last year because nothing was convincing enough), finally looks like I
found it :-)


Irio Lavagno
